ruby/ql/lib/codeql/ruby/ast/internal/Module.qll

```diff
@@ -495,23 +524,31 @@ private module ResolveImpl {
 private import ResolveImpl
 
 /**
- * A variant of AstNode::getEnclosingModule that excludes
+ * Gets an enclosing scope of `scope`, stopping at the first module or block.
+ *
+ * Includes `scope` itself and the final module/block.
+ */
+private Scope enclosingScopesNoBlock(Scope scope) {
```

Comment:
Why is it important to stop at block scopes? I'm not sure this change is right. For example:

```
A = "top"
module M
  A = "M"
end

M.module_eval { puts A }
```
prints `"top"`

Reply:
You're right. I'm not sure what I was thinking there. The block thing is specific to `include` calls and such since they rely on `self`.

Technically there was a bug in old implementation of `enclosingModule` since it relies on `getParent()`, although it would only affect really obscure cases like,
```rb
module (include A)::B
end
```
where the `include` call would be seen as being part of the `B` module, not the top-level. Still, it's a footgun that might reappear later so I think it's best to keep the new formulation.

I've pushed a commit to fix the place where we incorrectly relies on this predicate.